test/extensions/quic_listeners/quiche/test_utils.h

```diff
@@ -0,0 +1,57 @@
+#include "extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.h"
+
+#pragma GCC diagnostic push
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+// QUICHE uses offsetof().
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+
+#include "quiche/quic/core/http/quic_spdy_session.h"
+#include "quiche/quic/test_tools/quic_test_utils.h"
+#include "quiche/quic/core/quic_utils.h"
+
+#pragma GCC diagnostic pop
+
+namespace Envoy {
+namespace Quic {
+
+class MockEnvoyQuicSession : public quic::QuicSpdySession, public QuicFilterManagerConnectionImpl {
```

Comment:
Q: if MockEnvoyQuicSession inherits MockQuicSpdySession, can we remove the MOCK_METHODs?

Reply:
We don't want to inherit from MockQuicSpdySession because it uses quic::test::DefaultQuicConfig() which sets flow control window to very large value.